File: operation-spring/entity/dto/RealizedParcelOperation.java
package com.prompt.operation.entity.dto;

import com.prompt.operation.core.dto.OperationDto;

import java.time.LocalDateTime;

public class RealizedParcelOperation extends OperationDto {

    private int month;
    private int year;
    private byte[] monthlySum;
    private boolean approved;

    public RealizedParcelOperation(LocalDateTime occurredAt, int month, int year, byte[] monthlySum, boolean approved) {
        super(occurredAt);
        this.month = month;
        this.year = year;
        this.monthlySum = monthlySum;
        this.approved = approved;
    }

    public RealizedParcelOperation() {
    }

    public int getMonth() {
        return month;
    }

    public void setMonth(int month) {
        this.month = month;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public byte[] getMonthlySum() {
        return monthlySum;
    }

    public void setMonthlySum(byte[] monthlySum) {
        this.monthlySum = monthlySum;
    }

    public boolean isApproved() {
        return approved;
    }

    public void setApproved(boolean approved) {
        this.approved = approved;
    }
}

File: operation-nest/entity/realized-parcel-operation/realized-parcel-operation.ts
import { IsBase64, IsBoolean, IsInt, IsNotEmpty, Max, Min } from 'class-validator';
import { IsEncryptedNumber } from '../../../cryptography/is-encrypted-number';
import { OperationDto } from '../../core/domain/operation.dto';


export class RealizedParcelOperation extends OperationDto {

    @IsNotEmpty()
    @IsInt()
    @Min(1900)
    @Max(2100)
    year: number;

    @IsNotEmpty()
    @IsInt()
    @Min(1)
    @Max(12)
    month: number;

    @IsNotEmpty()
    @IsBase64()
    @IsEncryptedNumber()
    monthlySum: string;

    @IsNotEmpty()
    @IsBoolean()
    approved: boolean;

    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
    constructor(data?: any) {
        super(data);
        this.year = data && data.year || null;
        this.month = data && data.month || null;
        this.monthlySum = data && data.montlhySum || null;
        this.approved = data && data.approved || false;
    }

}